crates/mun_runtime/src/function.rs

```diff
@@ -83,3 +84,51 @@ impl FunctionInfoStorage {
         (fn_info, fn_storage)
     }
 }
+
+pub trait IntoFunctionInfo {
+    fn into<S: AsRef<str>>(
+        self,
+        name: S,
+        privacy: abi::Privacy,
+    ) -> (FunctionInfo, FunctionInfoStorage);
+}
+
+impl<R: ReturnTypeReflection> IntoFunctionInfo for extern "C" fn() -> R {
```

Comment:
We should turn this into a macro similar to `invoke_fn_impl`, because now it only supports functions up to 2 arguments and with a lot of code duplication.

Reply:
Done